include/bluetooth/conn.h

```diff
@@ -588,6 +589,15 @@ struct bt_conn_cb {
 	struct bt_conn_cb *_next;
 };
 
+/** @brief Update the initiation timeout for future connections
+ *
+ *  By default, the timeout is equal to
+ *  K_SECONDS(CONFIG_BT_CREATE_CONN_TIMEOUT)
+ *
+ *  @param timeout Updated initiation timeout
+ */
+void bt_conn_initiation_timeout(s32_t timeout);
```

Comment:
Providing this timeout would be better to do after the Advertising Extension API and move the timeout into this struct
https://github.com/zephyrproject-rtos/zephyr/pull/22013/files#diff-d4cf4d3017bb228fcdd63bde162014e5R295

Reply:
That certainly makes sense, its not like this will make it into a release before the advertising extensions. Is it likely that yourself or one of the other bluetooth contributors would add such functionality before the v2.3 release?

Reply:
Scanning that pull request, it doesn't look like the logic flow has changed for the path that this request affects, so I would probably be right to implement this on top of the extended advertising myself.